static/js/public/manage-snaps.js

```diff
@@ -0,0 +1,58 @@
+function getSelectedSnaps(snapCheckboxes) {
+  let snaps = [];
+
+  snapCheckboxes.forEach((snap) => {
+    if (snap.checked) {
+      snaps.push({
+        name: snap.dataset.snapName,
+      });
+    }
+  });
+
+  return snaps;
+}
+
+function updateSnaps() {
+  const manageSnapsForm = document.querySelector("#manage-snaps-form");
+  const saveButton = document.querySelector(".js-save-button");
+  const snapCheckboxes = Array.prototype.slice.call(
+    document.querySelectorAll(".js-snap-checkbox")
+  );
+
+  const originalSnapState = getSelectedSnaps(snapCheckboxes);
+  let currentSnaps = getSelectedSnaps(snapCheckboxes);
+  let dirtyData = false;
+
+  snapCheckboxes.forEach((snap) => {
+    snap.addEventListener("change", () => {
+      currentSnaps = getSelectedSnaps(snapCheckboxes);
```

Comment:
Maybe replace with a querySelectorAll with `:checked` as the function is doing DOM look ups anyway

Reply:
In this case we want data rather than DOM nodes. I've renamed the function so its intent is clearer.